kernel-compliance-check/src/lib.rs

```diff
@@ -0,0 +1,831 @@
+mod formatter;
+mod models;
+
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use eyre::{Context, Result};
+use futures::stream::{self, StreamExt};
+use hf_hub::api::tokio::{ApiBuilder, ApiError};
+use hf_hub::{Cache, Repo, RepoType};
+use kernel_abi_check::{check_manylinux, check_python_abi, Version};
+use object::Object;
+
+pub use formatter::Console;
+pub use models::*;
+
+pub use models::{AbiCheckResult, Cli, Commands, CompliantError, Format, Variant};
+
+// Import the generated code containing variant data and helper functions
+// This file is generated at compile time by build.rs and provides:
+//
+// - VARIANTS_DATA: Raw JSON string containing supported build variants
+// - get_variants(): Returns lazily parsed JSON data as &'static Value
+// - get_cuda_variants(): Returns all CUDA variants
+// - get_rocm_variants(): Returns all ROCm variants
+include!(concat!(env!("OUT_DIR"), "/variants_data.rs"));
+
+#[allow(clippy::too_many_lines)]
+async fn fetch_repository_async(
+    repo_id: &str,
+    revision: &str,
+    force_fetch: bool,
+    prefer_hub_cli: bool,
+) -> Result<()> {
+    println!("Repository: {repo_id} (revision: {revision})");
+
+    // Create API client
+    let api = ApiBuilder::from_env()
+        .high()
+        .build()
+        .context("Failed to create HF API client")?;
+
+    let repo = Repo::with_revision(repo_id.to_string(), RepoType::Model, revision.to_string());
+
+    // TODO: improve internal fetching logic to match cli speed/timeouts when downloading
+    // to avoid using huggingface-cli
+
+    // Attempt to fetch the repository using huggingface-cli
+    if prefer_hub_cli {
+        let huggingface_cli =
+            std::env::var("HUGGINGFACE_CLI").unwrap_or_else(|_| "huggingface-cli".to_string());
+
+        let mut cmd = std::process::Command::new(&huggingface_cli);
+        cmd.arg("download")
+            .arg(repo_id)
+            .arg("--revision")
+            .arg(revision);
+
+        if force_fetch {
+            cmd.arg("--force");
+        }
+
+        println!("Using huggingface-cli to download repository");
```

Comment:
NIT: I would print such messages to `stderr`. For me `stdout` is only for output that you may want to pipe to other processes.

Reply:
good point, ive updated many of the print lines to eprintln